src/jvmMain/kotlin/com/epam/drill/classloading/ClassLoadingUtil.kt

```diff
@@ -29,8 +29,16 @@ object ClassLoadingUtil {
 
     fun retrieveClassesData(config: String): String {
         val pp = PackagesPrefixes.serializer() parse config
-        val scanItPlease = ClassPath().scanItPlease(ClassLoader.getSystemClassLoader())
-        val filter = scanItPlease
+        val systemClassLoader = ClassLoader.getSystemClassLoader()
+        val classLoaders = getAllThreadsClassLoaders().filter { classLoader ->
+            classLoader.hierarchies(mutableListOf(classLoader)).contains(systemClassLoader)
+        }
+        val scanItPlease: Map<String, ClassLoader> =
+            classLoaders.map { cl -> ClassPath().scanItPlease(cl) }
+                .flatMap { it.entries }
+                .map { it.key to it.value }
+                .toMap()
+        val filter: Map<String, ClassLoader> = scanItPlease
             .filter { (classPath, _) ->
```

Comment:
try to use https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter-keys.html